internal/services/payment_to_submitter_service.go

```diff
@@ -84,82 +79,104 @@ func (s PaymentToSubmitterService) SendBatchPayments(ctx context.Context, batchS
 	return nil
 }
 
-// sendPaymentsReadyToPay sends SDP's ready-to-pay payments to the transaction submission service, inside a DB transaction.
-// Payments Ready to be sent means:
+// sendPaymentsReadyToPay sends SDP's ready-to-pay payments to the transaction submission service, using two DB
+// transactions (for SDP and TSS), in order to guarantee that the data is consistent in both data stores.
 //
-//	a. Payment is in `READY` status
-//	b. Receiver Wallet is in `REGISTERED` status
-//	c. Disbursement is in `STARTED` status
-func (s PaymentToSubmitterService) sendPaymentsReadyToPay(ctx context.Context, dbTx db.DBTransaction, tenantID string, payments []*data.Payment) error {
-	var transactions []txSubStore.Transaction
-	var failedPayments []*data.Payment
-	var pendingPayments []*data.Payment
-	for _, payment := range payments {
-		// 2. Validate that payments are ready to be sent
-		if errValidation := validatePaymentReadyForSending(payment); errValidation != nil {
-			// if payment is not ready for sending, we will mark it as failed later.
-			failedPayments = append(failedPayments, payment)
-			log.Ctx(ctx).Errorf("Payment %s is not ready for sending. Error:%s", payment.ID, errValidation.Error())
-			continue
-		}
-
-		// TODO: change TSS to use string amount [SDP-483]
-		amount, parseErr := strconv.ParseFloat(payment.Amount, 64)
-		if parseErr != nil {
-			return fmt.Errorf("parsing payment amount %s for payment ID %s: %w", payment.Amount, payment.ID, parseErr)
-		}
-		transaction := txSubStore.Transaction{
-			ExternalID:  payment.ID,
-			AssetCode:   payment.Asset.Code,
-			AssetIssuer: payment.Asset.Issuer,
-			Amount:      amount,
-			Destination: payment.ReceiverWallet.StellarAddress,
-			TenantID:    tenantID,
-		}
-		transactions = append(transactions, transaction)
-		pendingPayments = append(pendingPayments, payment)
-	}
-
-	// 3. Persist data in Transactions table
-	insertedTransactions, err := db.RunInTransactionWithResult(ctx, s.tssModel.DBConnectionPool, nil, func(tssDBTx db.DBTransaction) ([]txSubStore.Transaction, error) {
-		return s.tssModel.BulkInsert(ctx, tssDBTx, transactions)
+// Payments ready-to-pay meet all the following conditions:
+//
+//   - Payment is in `READY` status
+//   - Receiver Wallet is in `REGISTERED` status
+//   - Disbursement is in `STARTED` status.
+func (s PaymentToSubmitterService) sendPaymentsReadyToPay(
+	ctx context.Context,
+	tenantID string,
+	getPaymentsFn func(sdpDBTx db.DBTransaction) ([]*data.Payment, error),
+) error {
+	outerErr := db.RunInTransaction(ctx, s.sdpModels.DBConnectionPool, nil, func(sdpDBTx db.DBTransaction) error {
+		return db.RunInTransaction(ctx, s.tssModel.DBConnectionPool, nil, func(tssDBTx db.DBTransaction) error {
```

Comment:
I'm assuming the inside of this didn't change.

Reply:
You're right, it didn't change.